### PR TITLE
[AIRFLOW-5336] Add ability to make updating FAB perms on webserver in…

### PR DESCRIPTION
**Pull request by nritholtz**
…it optional

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

https://issues.apache.org/jira/browse/AIRFLOW-5336

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Add ability to make updating perms and other syncs on webserver init optional similar to https://github.com/dpgaspar/Flask-AppBuilder/pull/625
I am running into issues where multiple webserver processes are running what I believe is the sync_perm and it causes there to be records in the ab_permission_view with null permission_id (ultimately resulting in both processes crashing until I either manually clean them up or recreate the table). I envision being able to shut it off on the webserver init and use a one-off run that I am currently using, that handles it as part of deployment via the sync_perm CLI command.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Not sure if testing is needed here. If it is, let me know where to look.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg
Comment (feng-tao) [inline airflow/config_templates/default_airflow.cfg:346]: thanks for adding that. I think it will be helpful for other users to document when to turn on / off this flag, like in faq?

**Review by feng-tao — APPROVED** [review]

File: airflow/config_templates/default_airflow.cfg
Comment (feng-tao) [reply airflow/config_templates/default_airflow.cfg:346]: but the change LGTM.

Comment (kaxil) [reply airflow/config_templates/default_airflow.cfg:346]: @nritholtz agree with @feng-tao , can you please add some doc so that users know when to turn on / off this flag !

Comment (nritholtz) [reply airflow/config_templates/default_airflow.cfg:346]: @kaxil added entry to FAQ, PTAL.

File: docs/faq.rst
Comment (kaxil) [inline docs/faq.rst]: It feels unclear to me on what the issue is from this FAQ question but happy to merge this if @feng-tao thinks so !

Comment (ashb) [reply docs/faq.rst]: The null permission ID should have been fixed now via another change anyway, so this is more "how do I stop the sync perms happening multiple times per webserver" etc.

Comment (ashb) [reply docs/faq.rst]: #4764 was where the null perm was fixed.

Comment (nritholtz) [reply docs/faq.rst]: @ashb Updated the FAQ entry to `How do I stop the sync perms happening multiple times per webserver?`

**Review by kaxil — APPROVED** [review]
Thanks @nritholtz . Waiting for CI to turn green

File: docs/faq.rst
Comment (kaxil) [inline docs/faq.rst]:
```suggestion
-------------------------------------------------------------------
```

Comment (kaxil) [reply docs/faq.rst]: Ci fails without this

Comment (nritholtz) [reply docs/faq.rst]: Thanks @kaxil , applied.

File: docs/faq.rst
Comment (kaxil) [inline docs/faq.rst]:
```suggestion
--------------------------------------------------------------------
```

we missed 1 `-` :(

**Pull request merged**
